src/bin/lib/subcommands.c/runprogram.h

```diff
@@ -581,7 +585,20 @@ waitprogram(Program *prog, pid_t childPid)
 		}
 	} while (!WIFEXITED(status) && !WIFSIGNALED(status));
 
-	prog->returnCode = WEXITSTATUS(status);
+	if (WIFEXITED(status))
+	{
+		prog->returnCode = WEXITSTATUS(status);
+	}
+	else if (WIFSIGNALED(status))
+	{
+		int signo = WTERMSIG(status);
+		prog->returnCode = signo ? signo : -1;
```

Comment:
Why would we return -1 instead of 0, and why the signal number directly rather than what I believe the default for Unix is: 128 + Signo? I can be convinced that your current choice is the best one, with proper amount of comments and references. Thanks!

Reply:
Thank you for your suggestion. You're right.